src/Controller/Controller.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use Psr\Http\Message\ResponseInterface;
+use Yiisoft\Aliases\Aliases;
+use Yiisoft\View\ViewContextInterface;
+use Yiisoft\View\WebView;
+use Yiisoft\Yii\Web\Data\DataResponseFactoryInterface;
+
+use function is_file;
+use function pathinfo;
+
+abstract class Controller implements ViewContextInterface
+{
+    private string $layout;
+    private array $parameters = [];
+    private WebView $view;
+    protected Aliases $aliases;
+    protected DataResponseFactoryInterface $responseFactory;
+
+    public function __construct(
+        DataResponseFactoryInterface $responseFactory,
+        Aliases $aliases,
+        WebView $view
+    ) {
+        $this->responseFactory = $responseFactory;
+        $this->aliases = $aliases;
+        $this->view = $view;
+        $this->layout = $aliases->get('@layout') . '/main';
+    }
+
+    protected function render(string $view, array $parameters = []): ResponseInterface
```

Comment:
Please you can from https://github.com/yiisoft/yii-demo/blob/master/src/Controller.php#L34